src/javascripts/components/options-table.js

```diff
@@ -0,0 +1,44 @@
+import 'govuk-frontend/vendor/polyfills/Element/prototype/classList'
+
+var OptionsTable = {
+  init: function () {
+    OptionsTable.expandMacroOptions()
+  },
+  // Open Nunjucks tab and expand macro options details when URL hash is 'nunjucks-options-[example]'
+  expandMacroOptions: function () {
+    var hash = window.location.hash
+
+    if (hash.match('^#options-')) {
+      var exampleName = hash.split('#options-')[1]
+
+      if (exampleName) {
+        var tabLink = document.querySelector('a[href="#' + exampleName + '-nunjucks"]')
+        var tabHeading = tabLink ? tabLink.parentNode : null
+        var optionsDetailsElement = document.getElementById(hash.substring(1) + '-details')
+
+        if (tabHeading && optionsDetailsElement) {
+          var tabsElement = optionsDetailsElement.parentNode
+          var detailsSummary = optionsDetailsElement.querySelector('.govuk-details__summary')
+          var detailsText = optionsDetailsElement.querySelector('.govuk-details__text')
+
+          if (detailsSummary && detailsText) {
+            tabLink.setAttribute('aria-expanded', true)
+            tabHeading.className += ' app-tabs__item--current'
+            tabsElement.classList.remove('app-tabs__container--hidden')
+            tabsElement.setAttribute('aria-hidden', false)
+
+            optionsDetailsElement.setAttribute('open', 'open')
```

Comment:
Definitely not blocking – I wonder if we could look at using mutation observers on the details element, so that if the `open` attribute is changed programatically the ARIA attributes and styles always stay in sync?

Reply:
Yeah this code knows too much about the tabs and details component internals, so is likely to break since we're reimplementing logic that already exists.

For example the GOV.UK Frontend tabs listen to 'hashchange' events.

We could make this better after we merge this in though...

Reply:
Yes I agree, we should make this better after merging this.